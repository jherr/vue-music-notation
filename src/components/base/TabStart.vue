<template>
  <svg
    style="display: inline;"
    width="20"
    :height="(stringCount + 1) * lineHeight"
  >
    <line
      x1="15"
      x2="15"
      :y1="lines[0].y"
      :y2="lines[this.stringCount - 1].y"
      style="stroke-width: 1; stroke: black;"
    />
    <line
      x1="15"
      x2="20"
      :y1="line.y"
      :y2="line.y"
      style="stroke-width: 1; stroke: black;"
      v-for="line in lines"
      :key="`line_${line.y}`"
    />
    <text
      x="0"
      :y="line.y + (lineHeight / 4)"
      :style="`font-size: ${lineHeight * 0.8}px; font-family: Arial, Verdana, Sans-Serif;`"
      v-for="line in lines"
      :key="`tuning_${line.y}`"
    >
      {{ line.tuning }}
    </text>
  </svg>
</template>

<script>
import constants from '../constants';

export default {
  props: {
    stringCount: {
      default: 6,
    },
    tuning: {
      default: () => [ 'E', 'A', 'D', 'G', 'B', 'E' ],
    },
  },
  data() {
    const lines = [];
    for (let string = 0; string < this.stringCount; string++) {
      lines.push({
        y: (constants.tabulature.LINE_HEIGHT * (this.stringCount - string)) + (constants.tabulature.LINE_HEIGHT / 2),
        string,
        tuning: this.tuning[string],
      });
    }
    return {
      lines,
      lineHeight: constants.tabulature.LINE_HEIGHT,
    }
  }
}
</script>
