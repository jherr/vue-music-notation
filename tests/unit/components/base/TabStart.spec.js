import { mount } from '@vue/test-utils';
import TabStart from '@/components/base/TabStart.vue';

describe('TabStart', () => {
  it('renders', () => {
    const wrapper = mount(TabStart, {
      propsData: {
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
