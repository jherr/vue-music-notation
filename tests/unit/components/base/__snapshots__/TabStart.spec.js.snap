// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabStart renders 1`] = `
<svg
  height="105"
  style="display: inline;"
  width="20"
>
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="15"
    y1="97.5"
    y2="22.5"
  />
   
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="20"
    y1="97.5"
    y2="97.5"
  />
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="20"
    y1="82.5"
    y2="82.5"
  />
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="20"
    y1="67.5"
    y2="67.5"
  />
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="20"
    y1="52.5"
    y2="52.5"
  />
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="20"
    y1="37.5"
    y2="37.5"
  />
  <line
    style="stroke-width: 1; stroke: black;"
    x1="15"
    x2="20"
    y1="22.5"
    y2="22.5"
  />
   
  <text
    style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
    x="0"
    y="101.25"
  >
    
    E
  
  </text>
  <text
    style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
    x="0"
    y="86.25"
  >
    
    A
  
  </text>
  <text
    style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
    x="0"
    y="71.25"
  >
    
    D
  
  </text>
  <text
    style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
    x="0"
    y="56.25"
  >
    
    G
  
  </text>
  <text
    style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
    x="0"
    y="41.25"
  >
    
    B
  
  </text>
  <text
    style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
    x="0"
    y="26.25"
  >
    
    E
  
  </text>
</svg>
`;
